python/taichi/examples/simulation/karman_vortex_street.py

```diff
@@ -0,0 +1,176 @@
+            self.apply_bc_core(1, 0, 0, j, 1, j)
+
+            # right: dr = 2; ibc = nx-1; jbc = j; inb = nx-2; jnb = j
+            self.apply_bc_core(1, 2, self.nx - 1, j, self.nx - 2, j)
+
+        # top and bottom
+        for i in ti.ndrange(self.nx):
+            # top: dr = 1; ibc = i; jbc = ny-1; inb = i; jnb = ny-2
+            self.apply_bc_core(1, 1, i, self.ny - 1, i, self.ny - 2)
+
+            # bottom: dr = 3; ibc = i; jbc = 0; inb = i; jnb = 1
+            self.apply_bc_core(1, 3, i, 0, i, 1)
+
+        # cylindrical obstacle
+        # Note: for cuda backend, putting 'if statement' inside loops can be much faster!
+        for i, j in ti.ndrange(self.nx, self.ny):
+            if (self.cy == 1 and self.mask[i, j] == 1):
+                self.vel[i, j] = 0, 0  # velocity is zero at solid boundary
+                inb = 0
+                jnb = 0
+                if i >= self.cy_para[0]:
+                    inb = i + 1
+                else:
+                    inb = i - 1
+                if j >= self.cy_para[1]:
+                    jnb = j + 1
+                else:
+                    jnb = j - 1
+                self.apply_bc_core(0, 0, i, j, inb, jnb)
+
+    @ti.func
+    def apply_bc_core(self, outer, dr, ibc, jbc, inb, jnb):
+        if outer == 1:  # handle outer boundary
+            if self.bc_type[dr] == 0:
+                self.vel[ibc, jbc] = self.bc_value[dr]
+
+            elif self.bc_type[dr] == 1:
+                self.vel[ibc, jbc] = self.vel[inb, jnb]
+
+        self.rho[ibc, jbc] = self.rho[inb, jnb]
+        self.f_old[ibc, jbc] = self.f_eq(ibc, jbc) - self.f_eq(
+            inb, jnb) + self.f_old[inb, jnb]
+
+    def solve(self):
+        gui = ti.GUI('Karman Vortex Street', (self.nx, 2 * self.ny))
+        self.init()
+        while not gui.get_event(ti.GUI.ESCAPE, ti.GUI.EXIT):
+            for _ in range(10):
+                self.collide_and_stream()
+                self.update_macro_var()
+                self.apply_bc()
+
+            ##  code fragment displaying vorticity is contributed by woclass
+            vel = self.vel.to_numpy()
+            ugrad = np.gradient(vel[:, :, 0])
+            vgrad = np.gradient(vel[:, :, 1])
+            vor = ugrad[1] - vgrad[0]
+            vel_mag = (vel[:, :, 0]**2.0 + vel[:, :, 1]**2.0)**0.5
+            ## color map
+            colors = [(1, 1, 0), (0.953, 0.490, 0.016), (0, 0, 0),
+                      (0.176, 0.976, 0.529), (0, 1, 1)]
+            my_cmap = matplotlib.colors.LinearSegmentedColormap.from_list(
+                'my_cmap', colors)
+            vor_img = cm.ScalarMappable(norm=matplotlib.colors.Normalize(
+                vmin=-0.02, vmax=0.02),
+                                        cmap=my_cmap).to_rgba(vor)
+            vel_img = cm.plasma(vel_mag / 0.15)
+            img = np.concatenate((vor_img, vel_img), axis=1)
+            gui.set_image(img)
+            gui.show()
+
+    def pass_to_py(self):
```

Comment:
this  method is not used?

Reply:
Oh, yes it's an unused function. Removed.